### PR TITLE
terramate-bin: init at 0.8.3

### PR DESCRIPTION
**Pull request by asininemonkey**
## Description of changes

Latest binary release of Terramate: https://github.com/terramate-io/terramate

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by asininemonkey** [comment]
That was my original goal but there seems to be an issue with v0.4.3+ releases that prevents a successful build via Nix.

From what I can make of the error it appears to be related to a test requiring network access denied by the Nix build sandbox.

I'm not familiar enough with Go or `buildGoModule` to resolve the issue hence giving users an option to choose a binary release (especially for those using something `devbox` in CI/CD environments wanting to minimise runtimes).

**Comment by asininemonkey** [comment]
Would love to get a review so this can be merged and made available to Devbox users.

File: pkgs/by-name/te/terramate-bin/package.nix [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+  '';
+
+  meta = with lib; {
+    description = "Adds code generation, stacks, orchestration, change detection, data sharing and more to Terraform";
+    homepage = "https://github.com/terramate-io/terramate";
+    sourceProvenance = with sourceTypes; [ binaryNativeCode ];
+    license = licenses.mpl20;
+    maintainers = with maintainers; [ asininemonkey ];
+    mainProgram = "terramate";
+    platforms = [ "aarch64-darwin" "aarch64-linux" "i686-linux" "x86_64-darwin" "x86_64-linux" ];
```
Comment (Sigmanificient) [inline pkgs/by-name/te/terramate-bin/package.nix:67]:
```suggestion
    platforms = lib.platforms.linux ++ lib.platforms.darwin;
```

File: pkgs/by-name/te/terramate-bin/package.nix [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+  '';
+
+  installCheckPhase = ''
+    runHook preInstallCheck
+    $out/bin/terramate version
+    $out/bin/terramate-ls -version
+    runHook postInstallCheck
+  '';
+
+  meta = with lib; {
```
Comment (Sigmanificient) [inline pkgs/by-name/te/terramate-bin/package.nix:60]: Remove `with lib;`
See: https://github.com/NixOS/nixpkgs/issues/208242

File: pkgs/by-name/te/terramate-bin/package.nix [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+    runHook preInstall
+    install -D terramate $out/bin/terramate
+    install -D terramate-ls $out/bin/terramate-ls
+    runHook postInstall
+  '';
+
+  installCheckPhase = ''
+    runHook preInstallCheck
+    $out/bin/terramate version
+    $out/bin/terramate-ls -version
```
Comment (Sigmanificient) [inline pkgs/by-name/te/terramate-bin/package.nix:56]: maybe add a passthru version check?

**Review by Sigmanificient — CHANGES_REQUESTED** [review]

**Comment by Lagoja** [comment]
Result of `nix run nixpkgs#nixpkgs-review -- pr 293256` on aarch64 Darwin

1 package built:
terramate-bin

Seems to build and run fine on ARM Darwin

**Comment by funkeleinhorn** [comment]
> Would love to get a review so this can be merged and made available to Devbox users.

I do not think it makes sense to introduce this package as there is already an terramate package which builds from source and provides binaries through cachenix. I updated the exiting package to 0.8.3 in this PR: https://github.com/NixOS/nixpkgs/pull/310559 Please have a look if this solves your problem of an outdated version.

**Comment by asininemonkey** [comment]
Appreciate all the feedback and assistance with this PR. Will close in favour of PR [310559](https://github.com/NixOS/nixpkgs/pull/310559)

**Pull request closed without merging**